Gems/AWSCore/Code/Tools/ResourceMappingTool/utils/aws_utils.py

```diff
@@ -50,10 +50,14 @@ def _close_client_connection(client: BaseClient) -> None:
 
 
 def _initialize_boto3_aws_client(service: str, region: str = "") -> BaseClient:
-    if region:
-        boto3_client: BaseClient = default_session.client(service, region_name=region)
-    else:
-        boto3_client: BaseClient = default_session.client(service)
+    try:
+        if region:
+            boto3_client: BaseClient = default_session.client(service, region_name=region)
+        else:
+            boto3_client: BaseClient = default_session.client(service)
+    except (CredentialRetrievalError, PartialCredentialsError) as error:
```

Comment:
does this mean that other types of exceptions may not be raised if they're encountered? Are there other possible exception types we should consider adding, like [`UnknownCredentialError`](https://github.com/boto/botocore/blob/develop/botocore/exceptions.py#L435) or [`ProfileNotFound`](https://github.com/boto/botocore/blob/develop/botocore/exceptions.py#L252) ?

Reply:
Is there anything we should add here around providing help if/when this exception is raised? Any guidance we could add.

Reply:
Agree with the concern here. AWS documents don't list all the potential exceptions and I had to look into the botocore code to figure out other potential exceptions like PartialCredentialsError. I did some local tests on some common issues (like invalid profile or config file) and notice that we've handled the exceptions somewhere else in the resource mapping tool code, e.g. https://github.com/o3de/o3de/blob/development/Gems/AWSCore/Code/Tools/ResourceMappingTool/utils/aws_utils.py#L67.

It might not be the best practice to catch a general exception, but it seems like the only option here if we want to make sure that all the exceptions can be captured including new exception types added to botocore in the future.

Reply:
> Is there anything we should add here around providing help if/when this exception is raised? Any guidance we could add.

I think the error messages from botocore explain the condition when they're raised. For example, 

- For CredentialRetrievalError, the error message is "Error when retrieving credentials from {provider}: {error_message}"
- For PartialCredentialsError, the error message is "Partial credentials found in {provider}: missing: {cred_var}".

Reply:
Synced offline. Will catch the BotoCoreError exception which is the base class for all the botocore errors.